pkg/restic/restic.go

```diff
@@ -229,3 +229,42 @@ func SnapshotIDFromBackupLog(output string) string {
 	}
 	return ""
 }
+
+// SnapshotStatsFromStatsLog gets the Snapshot Stats from Stats Command log
+func SnapshotStatsFromStatsLog(output string) (string, string, string) {
+	if output == "" {
+		return "", "", ""
+	}
+	var fileCount string
+	var size string
+	logs := regexp.MustCompile("[\n]").Split(output, -1)
+	// Log should contain "Total File Count:   xx"
+	pattern1 := regexp.MustCompile(`Total File Count:\s+(.*?)$`)
+	// Log should contain "Total Size:   xx"
+	pattern2 := regexp.MustCompile(`Total Size: \s+(.*?)$`)
+	for _, l := range logs {
+		match1 := pattern1.FindAllStringSubmatch(l, 1)
+		if match1 != nil && match1[0] != nil {
+			fileCount = match1[0][1]
+		}
+		match2 := pattern2.FindAllStringSubmatch(l, 1)
+		if match2 != nil && match2[0] != nil {
+			size = match2[0][1]
+		}
+	}
+	return SnapshotStatsModeFromStatsLog(output), fileCount, size
+}
+
+// SnapshotStatsModeFromStatsLog gets the Stats mode from Stats Command log
+func SnapshotStatsModeFromStatsLog(output string) string {
+	logs := regexp.MustCompile("[\n]").Split(output, -1)
+	// Log should contain "Stats for .... in  xx mode"
+	pattern := regexp.MustCompile(`Stats for.*in\s+(.*?)\s+mode:`)
+	for _, l := range logs {
+		match := pattern.FindAllStringSubmatch(l, 1)
+		if match != nil && match[0] != nil {
```

Comment:
This still doesn't check the second lookup. Same for the ones above as well

```suggestion
		if len(match) > 0 && len(match[0]) > 1 {
```

Reply:
Ahh okay. I will update and be more specific